bin/manage.py

```diff
@@ -0,0 +1,406 @@
+    so that we can dynamically alter what service we advertise
+    """
+
+    def __init__(self):
+        # TODO: we should make sure we can support JSON-in-env-var
+        # the same as ContainerPilot itself
+        self.path = get_environ('CONTAINERPILOT', None).replace('file://', '')
+        with open(self.path, 'r') as f:
+            self.config = json.loads(f.read())
+
+    @debug
+    def update(self, state):
+        if state and self.config['services'][0]['name'] != state:
+            self.config['services'][0]['name'] = state
+            self.render()
+            return True
+
+    @debug
+    def render(self):
+        new_config = json.dumps(self.config)
+        with open(self.path, 'w') as f:
+            f.write(new_config)
+
+    def reload(self):
+        """ force ContainerPilot to reload its configuration """
+        log.info('Reloading ContainerPilot configuration.')
+        os.kill(1, signal.SIGHUP)
+
+# ---------------------------------------------------------
+# Top-level functions called by ContainerPilot or forked by this program
+
+@debug
+def pre_start():
+    """
+    MongoDB must be running in order to execute most of our setup behavior
+    so we're just going to make sure the directory structures are in
+    place and then let the first health check handler take it from there
+    """
+    # TODO is there anything that needs to be done before starting mongo?
+    sys.exit(0)
+
+@debug
+def health():
+    """
+    Run a simple health check. Also acts as a check for whether the
+    ContainerPilot configuration needs to be reloaded (if it's been
+    changed externally).
+    """
+    # TODO periodic mongodumps to Manta
+
+    hostname = socket.gethostname()
+    ip = get_ip()
+    local_mongo = MongoClient(ip, connect=False)
+
+    #  - check that mongo is responsive
+    if not is_mongo_up(local_mongo):
+        return False
+
+    # make sure this node has a valid consul session to work with
+    get_session()
+
+    #  - check for primary in replset and consul
+    is_mongo_primary = local_mongo.is_primary
+    consul_primary = get_primary_node_from_consul()
+
+    # no master in consul, and I am not master in mongo
+    if not consul_primary and not is_mongo_primary:
+        # check if this is already a replica set ['ok']
+        try:
+            repl_status = local_mongo.admin.command('replSetGetStatus')
+        except OperationFailure as e:
+            log.debug(e)
+            # this should only happen at the beginning when there is no replica set yet
+            # so the first node to get the lock in consul will initialize the set
+            # try setting self as primary in consul and rs.init()
+            mark_as_primary(hostname)
+            local_mongo.admin.command('replSetInitiate')
+    # otherwise let mongo elect a new master and it will mark itself in consul
+
+    #  if I am primary, make sure all nodes are in the replset config
```

Comment:
Should this all be in the `onChange` handler rather than the health check? We'll should see `onChange` events firing as each new instance comes up?